sklearn/isotonic.py

```diff
@@ -151,21 +152,29 @@ def isotonic_regression(
     array([2.75   , 2.75   , 2.75   , 2.75   , 7.33...,
            7.33..., 7.33..., 7.33..., 7.33..., 7.33...])
     """
-    order = np.s_[:] if increasing else np.s_[::-1]
     y = check_array(y, ensure_2d=False, input_name="y", dtype=[np.float64, np.float32])
-    y = np.array(y[order], dtype=y.dtype)
-    sample_weight = _check_sample_weight(sample_weight, y, dtype=y.dtype, copy=True)
-    sample_weight = np.ascontiguousarray(sample_weight[order])
+    if sp_version >= parse_version("1.12.0"):
```

Comment:
Typically, we add this type of branching in `utils.fixes`. Although, given the scope, I am okay with this approach.

Nit: It's a little bitter to compare base versions:

```suggestion
    if sp_base_version >= parse_version("1.12.0"):
```